paimon-core/src/main/java/org/apache/paimon/metrics/AbstractMetricGroup.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.metrics;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Contains key functionality for adding metrics and carries metrics.
+ *
+ * <p>A MetricGroup can be {@link #close() closed}. Upon closing, the group de-register all metrics.
+ *
+ * <p>The {@link #close()} method and {@link #addMetric(String, Metric)} method won't be invoked by
+ * multiple threads at the same time, {@link #addMetric(String, Metric)} and {@link #getMetrics()}
+ * have multi-threads problems, so they should be synchronized by the object lock.
+ */
+public abstract class AbstractMetricGroup implements MetricGroup {
+    protected static final Logger LOG = LoggerFactory.getLogger(MetricGroup.class);
+
+    // ------------------------------------------------------------------------
+
+    /** The map containing all tags and their associated values, lazily computed. */
+    protected Map<String, String> tags;
+
+    /** Flag indicating whether this group has been closed. */
+    private boolean closed = false;
+
+    private final Map<String, Metric> metrics = new HashMap<>();
+
+    private final String table;
+
+    // ------------------------------------------------------------------------
+
+    public AbstractMetricGroup(String table, @Nullable Map<String, String> tags) {
+        this.table = table;
+        this.tags = tags;
```

Comment:
In my option, `table` should exists in the metric name for users must concern about the metrics of one specific table, which means that people would rather filter a table first, and then query the metrics they need, but not query the metrics first and then specify a tag of `table`. How do you think?

Reply:
I disagree with this idea. If you take a look at the metric names in Flink, you will see that job id, operator id, etc., etc., are all tags. Many visualization software, for example Grafana, supports plotting the same metric with different tags at the same time. This is convenient for the users to quickly spot what table is behaving unexpectedly. If the metric names are different it would be difficult to plot all of them in the same graph.